### PR TITLE
Feat/lesq 729/legacy search [LESQ-729]

### PR DESCRIPTION
**Pull request by njblore**
## Description

Music year: 1953

- remove call to 2020 search api
- update legacy filter to exclude EYFS only
- remove all 2020 search api code
- update all tests to use 2023 fixture and refactor some functionality to be shared

## How to test

Once the search API changes have been deployed to production:

1. Go to https://deploy-preview-2357--oak-web-application.netlify.thenational.academy
2. Search for eg 'volcanoes'
3. You should see legacy results, once each (no duplicates)
4. in the requests tab you should only see requests to one search API
5. you should still see results for non-legacy content
6. if you search for eg 'veggies' you should get no results (EYFS content)

### Screenshots:
the results are coming back in a different order for legacy content, which suggests they're being weighted differently. Not sure we care at this point as it brings them in line with new content? Compare with prod vs this branch

Eg using legacy api:
Screenshot 2024-04-11 at 10 03 31

vs using new api:
Screenshot 2024-04-11 at 10 05 02

### CODE REVIEW

File: src/context/Search/search-api/2023/fetchResults.ts [hunk 1]
```diff
@@ -1,11 +1,50 @@
...
+          _source: {
+            ...source,
+            title,
+            slug,
+          },
+        };
+      }),
+    },
+  };
+  const snakeaseData = snakecaseKeys(transformedData, {
```
Comment (k-huggs) [inline src/context/Search/search-api/2023/fetchResults.ts:38]: not end of the world but small typo I think this should be snakecase

File: src/context/Search/useSearch.test.ts [hunk 2]
```diff
@@ -120,7 +102,7 @@ describe("useSearch()", () => {
 
     expect(fetch).toHaveBeenCalledTimes(1);
   });
-  test("results should be returned in the correct form", async () => {
+  test.skip("results should be returned in the correct form", async () => {
```
Comment (k-huggs) [inline src/context/Search/useSearch.test.ts:105]: Do we need a note as to why this test is being skipped?

**Review by k-huggs — APPROVED** [review]

**Comment by njblore** [comment]
@jack-skerrett-oak the qa testing is only on staging so won't be on prod, shall I ignore it anyway to be safe?
And yes agreed the computing subjects in particular are out of control

**Comment by njblore** [comment]
@jack-skerrett-oak that 'qa test' subject is only published on staging, right?

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
Bit of an odd one. The maths units now show the core tier, great. However, there is at least one example I have found where a [unit](https://www.thenational.academy/teachers/programmes/maths-secondary-ks4-core-l/units/simplifying-surds-71e6/lessons) that appears in both Core and Higher has two different search results, one for core and one for higher rather than them being in a drop down. Other units (like Directed numbers) correctly have the drop down for picking between core and foundation. 
Screenshot 2024-04-17 at 16 40 44

**Comment by jack-skerrett-oak** [comment]
> units

@njblore Ah OK, I didn't pick up on that. I guess it's fine for now though it does highlight we might need to be clearer on search results eventually

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**